cpp/rn-llama.cpp

```diff
@@ -719,4 +725,101 @@ std::vector<common_adapter_lora_info> llama_rn_context::getLoadedLoraAdapters()
     return this->lora;
 }
 
+bool llama_rn_context::initMultimodal(const std::string &mmproj_path) {
+    if (model == nullptr) {
+        LOG_ERROR("Model not loaded, cannot initialize multimodal", "");
+        return false;
+    }
+
+    // Initialize mtmd context
+    mtmd_context_params mtmd_params = mtmd_context_params_default();
+    mtmd_params.use_gpu = true;
+    mtmd_params.print_timings = true;
+    mtmd_params.n_threads = params.cpuparams.n_threads;
+    mtmd_params.verbosity = (lm_ggml_log_level)LM_GGML_LOG_LEVEL_INFO;
+
+    mtmd_ctx = mtmd_init_from_file(mmproj_path.c_str(), model, mtmd_params);
+    if (mtmd_ctx == nullptr) {
+        LOG_ERROR("Failed to initialize multimodal context with mmproj: %s", mmproj_path.c_str());
+        return false;
+    }
+
+    has_multimodal = true;
+    LOG_INFO("Multimodal context initialized successfully with mmproj: %s", mmproj_path.c_str());
+    return true;
+}
+
+bool llama_rn_context::processImage(
+    const std::string &image_path,
+    const std::string &prompt
+) {
+    if (!has_multimodal || mtmd_ctx == nullptr) {
+        LOG_ERROR("Multimodal context not initialized", "");
+        return false;
+    }
+
+    // Load the image from file
+    mtmd_bitmap* bitmap = mtmd_helper_bitmap_init_from_file(image_path.c_str());
+    if (bitmap == nullptr) {
+        LOG_ERROR("Failed to load image: %s", image_path.c_str());
+        return false;
+    }
+
+    // Create input chunks
+    mtmd_input_chunks* chunks = mtmd_input_chunks_init();
+    if (chunks == nullptr) {
+        LOG_ERROR("Failed to initialize input chunks", "");
+        mtmd_bitmap_free(bitmap);
+        return false;
+    }
+
+    // Create input text
+    mtmd_input_text input_text;
+    input_text.text = prompt.c_str();
+    input_text.add_special = true;
```

Comment:
```suggestion
    input_text.add_special = n_past == 0;
```

In case we're evaluating the first chunk in the conversation, we may want to add a BOS token to the beginning